surefire-api/src/main/java/org/apache/maven/surefire/api/booter/ForkingRunListener.java

```diff
@@ -68,7 +68,7 @@ public void testSetStarting( TestSetReportEntry report )
     @Override
     public void testSetCompleted( TestSetReportEntry report )
     {
-        target.systemProperties( report.getSystemProperties() );
+        target.systemProperties( report.getSystemProperties(), null, null );
```

Comment:
In this class we have filed `runMode` 
 - why not used here?
 - why is not propagated to `target`?

Reply:
I did split the entire work of [SUREFIRE-1860](https://issues.apache.org/jira/browse/SUREFIRE-1860) into 5 pieces.
The line you are asking for is in my IDEA:
`target.systemProperties( report.getSystemProperties(), report.getRunMode(), report.getTestRunId() );`
Both getters will come in the next PR right after - [SUREFIRE-2015](https://issues.apache.org/jira/browse/SUREFIRE-2015)

Reply:
The `target` is not JUnit's RunListener implementation.
Only the framework's listener may have this data and be "stateful".
The `target` is a pure encoder, and so it should not have any notion about stateful data regarding test status. It should only receive data, encode it, and send to a channel.

Reply:
So what for `EventChannelEncoder` has `runMode` field?

Reply:
`runMode` field will be removed in the next PR.
This PR only changed some method signatures the same as the previous one.
After this the implementation would come in PR.